manim/animation/animation.py

```diff
@@ -192,29 +300,92 @@ def get_sub_alpha(self, alpha: float, index: int, num_submobjects: int) -> float
 
     # Getters and setters
     def set_run_time(self, run_time: float) -> "Animation":
+        """Set the run time of the animation.
+
+        Parameters
+        ----------
+        run_time
+            The new time the animation should take in seconds.
+
+        .. note::
+
+            The run_time of an animation should not be changed while it is already
+            running.
+
+        Returns
+        -------
+        ``Animation``
+            ``self``
+        """
         self.run_time = run_time
         return self
 
     def get_run_time(self) -> float:
+        """Get the run time of the animation.
+
+        Returns
+        -------
+        float
+            The time the animation takes in seconds.
+        """
         return self.run_time
 
     def set_rate_func(
         self,
         rate_func: Callable[[float], float],
     ) -> "Animation":
+        """Set the rate function of the animation.
+
+        Parameters
+        ----------
+        run_time
+            The new time the animation should take in seconds.
+
+
+        Returns
+        -------
+        ``Animation``
+            ``self``
+        """
         self.rate_func = rate_func
         return self
 
     def get_rate_func(
         self,
     ) -> Callable[[float], float]:
+        """Get the rate function of the animation.
+
+        Returns
+        -------
+        Callable[[float], float]
+            The rate fucntion of the animation.
+        """
         return self.rate_func
 
     def set_name(self, name: str) -> "Animation":
+        """Set the name of the animation.
+
+        Parameters
+        ----------
+        name
+            The new name of the animation.
+
+        Returns
+        -------
+        ``Animation``
```

Comment:
use 
```suggestion
        :class:`Animation`
```

Reply:
this needs to be changed everywhere

Reply:
I feel like this is a stylistic choice we should discuss in general - do we really want to link to the page we are already on in every other sentence? I find it a bit unnecessary and cluttered to do that...

Reply:
Or you can remove the ` there. https://numpydoc.readthedocs.io/en/latest/format.html#returns
```suggestion
        Animation
```

Reply:
> Or you can remove the ` there

I did that :)